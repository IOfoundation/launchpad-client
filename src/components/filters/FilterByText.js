import React from 'react';
import {PropTypes} from 'prop-types';
import {MdSearch} from 'react-icons/lib/md';
import {MdClear} from 'react-icons/lib/md';
import Chip from '../filters/Chip';
import {isEmpty, isString} from 'lodash';

class FilterByText extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      labelTop: false,
      showFilterLabel: false,
      showDropdown: false,
      value: '',
    };
  }
  deleteFilter(e) {
    const filter = e.currentTarget.getAttribute('data-value');
    this.props.handleOnRemoveFilterOption(filter);
  }

  clearAll() {
    this.setState({showFilterLabel: false, labelTop: false});
    this.props.handleClickOnClearAllFilters();
  }

  _inputClicked() {
    this.setState({labelTop: true});
  }

  _closeSearch() {
    this.setState({labelTop: false});
  }

  handleKeyPress(value) {
    this.setState({value: event.target.value, showDropdown: true});
    this.props.handleTextSearchBusinesses(value);
    if (isEmpty(value)) {
      this.setState({showDropdown: false});
    }
  }

  handleClickOutside() {
    this.setState({labelTop: false});
  }

  renderFilter() {
    const filters = this.props.getFilterChips();
    if (isEmpty(filters.category)) {
      return null;
    } else if (isString(filters.category)) {
      return (
        <Chip
          key={filters.category}
          text={filters.category}
          handleClick={this.deleteFilter.bind(this)}
        />
      );
    }
    return filters.category.map(filter => (
      <Chip
        key={filter}
        text={filter}
        handleClick={this.deleteFilter.bind(this)}
      />
    ));
  }

  renderDropdown() {
    return (
      <ul className="option-dropdown-list">
        {this.props.search_results.map(search_result => (
          <li key={search_result.id}>
            <a
              href={search_result.searchable_type === 'id' &&
                `/businesses?id=${search_result.searchable_id}`
              }
            >
              {search_result.content}
            </a>
          </li>
        ))}
      </ul>
    );
  }

  render() {
    const filters = this.props.getFilterChips();
    return (
      <div
        className={
          this.state.labelTop ? (
            'col-md-12 col-xs-12 text-xs-margin filterTextContainer noPadding m-bot-16'
          ) : (
            'col-md-12 col-xs-12 text-xs-margin filterTextContainer noPadding'
          )
        }
      >
        <div className="grid search-text-form p-bot-16">
          <div
            className={
              isEmpty(filters) || this.state.labelTop ? (
                'filter-label-container-hide'
              ) : (
                'filter-label-container-show col-lg-9 noPadding'
              )
            }
          >
            {this.renderFilter()}
            <a className="search-filter-clear" onClick={() => this.clearAll()}>
              {'Clear All'}
            </a>
          </div>
          <h3
            className={
              isEmpty(filters) ? (
                'show-filter col-lg-9 p-left-0'
              ) : (
                'col-lg-9 hide-filter'
              )
            }
          >
            {'Filter results with the selections below'}
          </h3>
          <div
            className={
              this.state.labelTop ? (
                'full-width-text-filter'
              ) : (
                'col-lg-3 noPadding'
              )
            }
          >
            <input
              type="text"
              value={this.state.value}
              onClick={() => this._inputClicked()}
              onKeyUp={e => this.handleKeyPress(e.target.value)}
              placeholder={
                this.state.labelTop ? (
                  'Search by Resource Name or Ipsum'
                ) : (
                  'Or search by name'
                )
              }
              className={
                this.state.labelTop ? (
                  'full-width-search'
                ) : (
                  'text-search small-width-search'
                )
              }
            />
            {this.state.labelTop && (
              <MdClear
                className="text-search-icon"
                size="32"
                color="#2AD587"
                onClick={() => this._closeSearch()}
              />
            )}
          </div>
          <div
            className={
              this.state.showDropdown ? 'option-dropdown-show' : 'hero_input-hide'
            }
          >
            {this.renderDropdown()}
          </div>
          {!this.state.labelTop && (
            <MdSearch className="text-search-icon" size="32" color="#2AD587" />
          )}
        </div>
      </div>
    );
  }
}

FilterByText.propTypes = {
  getFilterChips: PropTypes.func.isRequired,
  handleClickOnClearAllFilters: PropTypes.func.isRequired,
  handleOnRemoveFilterOption: PropTypes.func.isRequired,
  handleTextSearchBusinesses: PropTypes.func.isRequired,
  search_results: PropTypes.arrayOf(PropTypes.object),
};

export default FilterByText;
