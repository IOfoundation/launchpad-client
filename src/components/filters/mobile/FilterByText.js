import React, {Component} from 'react';
import {PropTypes} from 'prop-types';
import {MdSearch} from 'react-icons/lib/md';
import Chip from '../../shared/Chip';
import {isEmpty, isString} from 'lodash';

class FilterByTextMobile extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchText: '',
      showDropdown: false,
    };
  }
  deleteFilter(e) {
    const filter = e.currentTarget.getAttribute('data-value');
    this.props.handleOnChangeFilterOptions(filter, 'category', true);
  }

  clearAll() {
    this.setState({showFilterLabel: false, labelTop: false});
    this.props.handleClickOnClearAllFilters();
  }

  handleDropdownOnClick(item) {
    item.searchable_type === 'Category' ? (
      this.props.handleOnChangeFilterOptions(item.content, 'category')
    ) : (
      this.props.handleOnChangeFilterOptions(item.searchable_id, 'organization')
    );
    this.setState({showDropdown: false, value: ''});
  }

  handleKeyPress(e) {
    const value = e.target.value;
    this.setState({searchText: value, showDropdown: true});
    this.props.getTextSearchResults(value);
    if (isEmpty(value)) {
      this.setState({showDropdown: false});
    }
  }
  renderFilter() {
    const filters = this.props.getFilterChips();
    if (isEmpty(filters.category)) {
      return null;
    } else if (isString(filters.category)) {
      return (
        <Chip
          key={filters.category}
          text={filters.category}
          handleClick={(e) => this.deleteFilter(e)}
          canDelete={true}
        />
      );
    }
    return filters.category.map(filter => (
      <Chip
        key={filter}
        text={filter}
        handleClick={e => this.deleteFilter(e)}
        canDelete={true}
      />
    ));
  }

  renderDropdown() {
    return (
      <ul className="option-dropdown-list">
        {this.props.items.map(item => (
          <li key={item.id}>
            <a onClick={e => this.handleDropdownOnClick(item, e)}>
              {item.content}
            </a>
            {item.searchable_type === 'Organization' ? (
              <img src="../static-data/images/organization.png" />
            ) : null}
          </li>
        ))}
        {!this.props.items.length && (
          <li className="not-match-message">
            <a>{'This does not match any item on our platform'}</a>
          </li>
        )}
      </ul>
    );
  }

  render() {
    const filters = this.props.getFilterChips();
    return (
      <div className="filter-chip">
        <div className="p-left-16 p-right-16">
          <h3 className="col-lg-7 noPadding">
            {'Filter results with the selections below'}
          </h3>
          {this.renderFilter()}
          <a className="search-filter-clear" onClick={() => this.clearAll()}>
            {'Clear All'}
          </a>
        </div>
        <div className="search-input-container p-left-16 p-right-16 m-top-16">
          <input
            type="text"
            value={this.state.searchText}
            onChange={e => this.handleKeyPress(e)}
            className="text-search full-width-search"
            placeholder="Or search by name"
          />
          <MdSearch className="text-search-icon" size={24} color="#2AD587" />
        </div>
        <div
          className={
            this.state.showDropdown ? 'option-dropdown-show' : 'hero_input-hide'
          }
        >
          {this.renderDropdown()}
        </div>
      </div>
    );
  }
}
FilterByTextMobile.PropTypes = {
  getFilterChips: PropTypes.func.isRequired,
  getTextSearchResults: PropTypes.func.isRequired,
  handleClickOnClearAllFilters: PropTypes.func.isRequired,
  handleOnChangeFilterOptions: PropTypes.func.isRequired,
  handleDropdownOnClick: PropTypes.func.isRequired,
};
export default FilterByTextMobile;
