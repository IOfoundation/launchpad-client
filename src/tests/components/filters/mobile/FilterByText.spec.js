import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import FilterByText from 'components/filters/mobile/FilterByText';
import organizationFixture from 'tests/fixtures/organization';

const props = {
  appliedFilters: {
    page: 1,
  },
  filterById: false,
  getTextSearchResults: jest.fn(),
  handleClickOnClearAllFilters: jest.fn(),
  handleOnChangeFilterOptions: jest.fn(),
  items: [],
  organizations: [organizationFixture],
};

describe('<FilterByText />', () => {
  it('renders snapshot for mobile FilterByText component', () => {
    const wrapper = shallow(<FilterByText {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
