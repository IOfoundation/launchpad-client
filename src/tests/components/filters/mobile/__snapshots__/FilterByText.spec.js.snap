// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterByText /> renders snapshot for mobile FilterByText component 1`] = `
<div
  className="filter-chip"
>
  <div
    className="search-input-container p-left-16 p-right-16"
  >
    <input
      className="search-by-text text-thin"
      onChange={[Function]}
      placeholder="Or search by name"
      type="text"
      value=""
    />
    <img
      className="search-by-text-icon"
      src="/static-data/images/search.png"
    />
  </div>
  <div
    className="hero_input-hide"
  >
    <ul
      className="option-dropdown-list"
    >
      <li
        className="not-match-message"
      >
        <a>
          This does not match any item on our platform
        </a>
      </li>
    </ul>
  </div>
</div>
`;
